### PR TITLE
Test latest version

### PR DESCRIPTION
**Pull request by ocefpaf**
The build system changed a lot and we will need to change it here to reflect that.

### CODE REVIEW

**Comment by ocefpaf** [comment]
@hobu could you review this one? I'm not sure I got everything right with cmake there.

**Comment by hobu** [comment]
We need parts of those windows patches.

**Comment by ocefpaf** [comment]
> We need parts of those windows patches.

Looks like `tool.h.patch` is upstream and I'm not sure we really need `cmakelists.patch`, do we? (I'm not a Windows expert though.)

Do you have any idea what may be triggering the Windows failure?

@gillins can you take a look at our Windows builds here?

File: recipe/bld.bat [hunk 1]
```diff
@@ -1,5 +1,4 @@
-mkdir build
-cd build
+mkdir build && cd build
 
 cmake -G "%CMAKE_GENERATOR%" ^
       -D CMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX% ^
```
Comment (hobu) [inline recipe/bld.bat]: After the cmake, you actually need to `ninja` to build the package.

**Comment by hobu** [comment]
I think the cure for https://ci.appveyor.com/project/conda-forge/libspatialindex-feedstock/builds/23111393/job/brpglf0kfjiyycnr#L648 was in that patch

**Comment by ocefpaf** [comment]
> I think the cure for [ci.appveyor.com/project/conda-forge/libspatialindex-feedstock/builds/23111393/job/brpglf0kfjiyycnr#L648](https://ci.appveyor.com/project/conda-forge/libspatialindex-feedstock/builds/23111393/job/brpglf0kfjiyycnr#L648) was in that patch

Actually, I think we should just drop `vc9`. No need to support something for `Python 2.7`.

**Comment by gillins** [comment]
> I'm not sure we really need cmakelists.patch

No, seems they have fixed this upstream also.

**Comment by hobu** [comment]
“They” is me. I tried to upstream the patches I saw here before cutting the release.

**Comment by ocefpaf** [comment]
@hobu to avoid issues like the `libnetcdf 4.6.3` and `giflib 5.1.7` let's try to double check the ABI before merging this.

Based on [1] the ABI in the"git" version is not 100% compatible with `1.8.5`. Do you know if that reflects the `1.9.0` or the compatibility increased/decreased?

Currently we are not pinning b/c `libspatialindex` was not updated in ages. Should we pin to major, minor, or micro?

[1] http://upstream.rosalinux.ru/versions/libspatialindex.html

**Comment by hobu** [comment]
> Currently we are not pinning b/c libspatialindex was not updated in ages. Should we pin to major, minor, or micro?

1.9.0 was mostly a roll up of a bunch of bug fixes, but there were some API modifications which necessitated doing a minor release. This included addition of `virtual` to a few member functions of the index types in the C++ API. I think the pin should probably be to the minor release, but most of the libspatialindex users use the C API which had no substantive changes. Full diff is [here](https://github.com/libspatialindex/libspatialindex/compare/9ce9e86c27b89f17e8b10e1651edbcfc46d1d34d..175a5a3f560c3c4c0c00d797890491274a8b1688)

**Comment by ocefpaf** [comment]
I added a `pin_subpackage` in https://github.com/conda-forge/libspatialindex-feedstock/pull/18/commits/a000a8e6fabcc45cef330af70b3987f2fed9e5a1 and a global pin in https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/211

Still, b/c it was unpinned in the past we will probably break everything that was build with `libspatialindex 1.8.5`. I guess we need a repo patch before merging this one or a swift rebuild and removal of the stale packages.

**Comment by ocefpaf** [comment]
@conda-forge-admin, please rerender

**Pull request merged**